### PR TITLE
Allow passing extra environment variables to PuppetCommand.format_command

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: lib/kafo/puppet_command.rb [hunk 1]
```diff
@@ -50,11 +50,11 @@ def self.aio_puppet?
       false
     end
 
-    def self.format_command(command)
+    def self.format_command(command, extra_env = {})
       if aio_puppet?
```
Comment (ekohl) [inline lib/kafo/puppet_command.rb:54]: Perhaps refactor this method to
```ruby
env = aio_puppet? clean_env_vars : ::ENV.to_h
[env.merge(extra_env), command, :unsetenv_others => false]
```

**Comment by ekohl** [comment]
Note in https://github.com/theforeman/kafo/pull/374 I also started to add tests. Should we build on that?

File: lib/kafo/puppet_command.rb (hunk 1 above)
Comment (evgeni) [reply lib/kafo/puppet_command.rb:54]: Considered, but `unsetenv_others` is also different in those branches.

Comment (ekohl) [reply lib/kafo/puppet_command.rb:54]: Ah yes, now I see why I wrote it that way in the past.

**Comment by evgeni** [comment]
> Note in #374 I also started to add tests. Should we build on that?

Sounds like a plan! You want to (quickly?) finish up #374, or shall I copy over the tests here?

**Comment by ekohl** [comment]
I just rebased it to make RuboCop green.

**Pull request merged**